### ISSUE TITLE
Ignore compile errors when generating into same package

### ISSUE THREAD

**Issue by jmattheis** [jmattheis/goverter#111]
Goverter requires type information for generating the conversion methods. When `output:file` is the defined in the same package as the conversion interface it likely that the generated content will produce compile errors if the source or target types are changed.

To properly work around this issue, you currently have to remove the generated file before running goverter. Example: https://github.com/hetznercloud/hcloud-go/blob/701f3810de06eeffbb6d820abfd6d0c7d6ad5ec1/script/generate_schema.sh

With the addition of #77 this problem will be more likely because it more or less requires that the generated output is in the same package.

Goverter should natively support by ignoring files generated by goverter when loading the packages for conversion. I've two proposals for this:
1. Remove the generated files before loading packages. Either by removing all files with the `generated by github.com/jmattheis/goverter ...` comment at the top, or by removing all files defined by `output:file`.
2. Add a [build constraint](https://pkg.go.dev/cmd/go#hdr-Build_constraints) to all generated files. E.g.
   ```
   //go:build !goverter_scan
   // +build !goverter_scan
   ```
   and then load the conversion interface with `-tags goverter_scan`. This should prevent compile errors originating from goverter generated files.

I'm in favor of the second proposal.

**Comment by jmattheis** [comment]
@pauhull do you have an opinion regarding this issue?

**Comment by jmattheis** [comment]
> Maybe there could be a goverter flag (something like assignImplInstanceTo <variable name>) to include the init function in the generated file?

I think this is too specific to your use-case, but I'd be okay with a more generic solution like adding raw code to the `output:file`:
```go
// goverter:converter
// goverter:output:file ./output.gen.go
// goverter:output:raw func init() {
// goverter:output:raw     converter = &converterImpl{}
// goverter:output:raw }
type converter interface {
}
```

**Comment by pauhull** [comment]
Yes, that makes sense. I don't know if it's possible, but maybe a solution like this would look cleaner:
```go
// goverter:converter
// goverter:output:file ./output.gen.go
/* goverter:output:raw 
func init() {
    converter = &converterImpl{}
}*/
type converter interface {
}
```

**Comment by jmattheis** [comment]
That will be difficult because the current setting only uses lines starting with `goverter:`. I've moved this into a separate ticket #113 as the solution with build constraints (https://github.com/jmattheis/goverter/issues/111#issuecomment-1835831040) should be good enough for now.

**Comment by jmattheis** [comment]
@pauhull could you try out the `ignore-compile-error` branch? Changes: https://github.com/jmattheis/goverter/compare/ignore-compile-error?expand=1

With this, generated files will receive `go:build !goverter` as build directive and when loading packages `-tags goverter` is satisfied.

- `go install github.com/jmattheis/goverter/cmd/goverter@ignore-compile-error`
- or `go run github.com/jmattheis/goverter/cmd/goverter@ignore-compile-error [pattern]`
- or `go get github.com/jmattheis/goverter@ignore-compile-error`

**Comment by pauhull** [comment]
Yes, the build flags and type assignment now work with the branch you provided 🙂

**Comment by jmattheis** [comment]
@pauhull Awesome, thanks for trying it out!

**Comment by jmattheis** [comment]
Fixed in [v1.3.0](https://goverter.jmattheis.de/changelog). See https://goverter.jmattheis.de/reference/build-constraint
